### PR TITLE
Handle 'optimal' when validating CCE error message parsing

### PR DESCRIPTION
**Pull request by tilne**
In order to make sure the supported instance types and families parsed
from the Batch's CCE are sane, they are compared to the list of all
known instance types and families returned by EC2's
DescribeInstanceTypeOfferings API.

This check was not accounting for the fact that Batch users can specify
an instance type/family of `optimal`. This change fixes that.

Signed-off-by: Tim Lane <tilne@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/pcluster/utils.py [hunk 1]
```diff
@@ -359,9 +359,14 @@ def _get_instance_families_from_types(instance_types):
     return list(families)
 
 
-def _instance_types_and_families_are_supported(candidate_types_and_families, known_types_and_families):
-    """Return a boolean describing whether the given instance types and families are known to be supported."""
-    unknowns = [candidate for candidate in candidate_types_and_families if candidate not in known_types_and_families]
+def _batch_instance_types_and_families_are_supported(candidate_types_and_families, known_types_and_families):
+    """Return a boolean describing whether the instance types and families parsed from Batch API are known."""
+    known_exceptions = ["optimal"]
```
Comment (demartinofra) [inline cli/pcluster/utils.py:364]: optimal can't be specified in a comma separated list but it needs to be the only value provided right?

Comment (tilne) [reply cli/pcluster/utils.py:364]: that wasn't a constraint i was aware of. is that enforced by the Batch API?

i'll verify whether or not that's the case now.

Comment (tilne) [reply cli/pcluster/utils.py:364]: I was able to successfully create a cluster using `compute_instance_type = optimal,m3.xlarge`, so at least the compute environment can be created using those multiple values. I seem to be able to submit jobs as well.

**Comment by tilne** [comment]
Amended the commit to add a log message with the instance types/families that were actually parsed from the CCE error message.

**Review by demartinofra — APPROVED** [review]

**Pull request merged**